### PR TITLE
Switched DownloadBuildArtifactV0 to agent plugin. Aligned DownloadBuildArtifactV1 task with V0

### PR DESCRIPTION
**Pull request by anatolybolshakov**
**Task name**: DownloadBuildArtifactV1, DownloadBuildArtifactV0

**Description**: switched DownloadBuildArtifactV0 to agent plugin. Aligned inputs for DownloadBuildArtifactV1 and DownloadBuildArtifactV0. Bumped minimum agent version for these tasks to 2.191.1 - to align with plugin changes.

**Documentation changes required:** Y (in progress)

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected - tested for V0/V1 tasks with simple scenarios

### CODE REVIEW

**Review by DaniilShmelev — APPROVED** [review]

**Review by EzzhevNikita — COMMENTED** [review]
I think we should remove ts implementation of the task from DownloadBuildArtifactsV0, since it won't be used in fact. Do you have any concerns about it?

File: Tasks/DownloadBuildArtifactsV1/task.json [hunk 1]
```diff
@@ -197,6 +197,14 @@
             "groupName": "advanced",
             "required": false,
             "helpMarkDown": "Optional number of times to retry downloading a build artifact if the download fails."
+        },
+        {
+            "name": "extractTars",
```
Comment (alexander-smolyakov) [inline Tasks/DownloadBuildArtifactsV1/task.json:202]: Considering the following pull request: https://github.com/microsoft/azure-pipelines-agent/pull/3462
I believe we also should add `cleanDestinationFolder` input.

https://github.com/microsoft/azure-pipelines-tasks/blob/71f8ac83bdd6a4838313fc54c32d5f2fb17fc2b7/Tasks/DownloadBuildArtifactsV0/task.json#L173-L179

**Review by alexander-smolyakov — CHANGES_REQUESTED** [review]

File: Tasks/DownloadBuildArtifactsV1/task.json (hunk 1 above)
Comment (anatolybolshakov) [reply Tasks/DownloadBuildArtifactsV1/task.json:209]: Fixed thanks!

**Review by alexander-smolyakov — APPROVED** [review]

**Comment by anatolybolshakov** [comment]
> I think we should remove ts implementation of the task from DownloadBuildArtifactsV0, since it won't be used in fact. Do you have any concerns about it?

Yes make sense, let me update also

**Review by EzzhevNikita — APPROVED** [review]

**Comment by EzzhevNikita** [comment]
LFTM! Tested these changes manually found no issues, everything work as expected

**Pull request merged**